### PR TITLE
Examples unification

### PR DESCRIPTION
**Pull request by GTVolk**
Hello, i'am unificated examples for same table structure

### CODE REVIEW

File: examples/read-only-auth/docker-compose.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@ services:
   registry:
     image: registry:2.7
     volumes:
-      - ./registry-data:/var/lib/registry
-      - ./registry.yml:/etc/docker/registry/config.yml
+      - ./data/registry:/var/lib/registry:rw
+      - ./conf/registry/config.yml:/etc/docker/registry/config.yml:ro
```
Comment (Joxit) [inline examples/read-only-auth/docker-compose.yml:7]: The real filename is still `registry.yml` and not `config.yml`

File: examples/token-auth-keycloak/conf/proxy/nginx.conf [hunk 2]
```diff
@@ -21,58 +23,52 @@ server {
...
-    location /ui {
-      proxy_pass $ui;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header Host               $host;
+        proxy_set_header X-Forwarded-Host   $host;
+        if ($http_authorization = "") {
+            add_header WWW-Authenticate 'Basic realm="Keycloak login"' always;
+            return 401;
+        }
+        proxy_pass $keycloak;
```
Comment (Joxit) [inline examples/token-auth-keycloak/conf/proxy/nginx.conf:61]: I use 2 spaces and not 4

File: examples/token-auth-keycloak/conf/proxy/nginx.conf [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
     set $registry "http://registry:5000";
     set $ui "http://ui";
 
+    charset utf-8;
 
-    #charset koi8-r;
-    #access_log  /var/log/nginx/host.access.log  main;
+    rewrite_log on;
+    access_log /dev/stdout main;
+    error_log /dev/stdout debug;
```
Comment (Joxit) [inline examples/token-auth-keycloak/conf/proxy/nginx.conf:14]: This is an example, please keep this simple and do not add unnecessary configuration

File: examples/token-auth-keycloak/conf/proxy/nginx.conf [hunk 4]
```diff
@@ -21,58 +23,52 @@ server {
...
 
     location /auth {
-      proxy_set_header X-Forwarded-Proto $scheme;
-      proxy_set_header Host               $host;
-      proxy_set_header X-Forwarded-Host   $host;
-      proxy_pass $keycloak;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header Host               $host;
+        proxy_set_header X-Forwarded-Host   $host;
+        proxy_pass $keycloak;
```
Comment (Joxit) [inline examples/token-auth-keycloak/conf/proxy/nginx.conf:50]: I use 2 spaces and not 4

**Review by Joxit — CHANGES_REQUESTED** [review]
Hi, thank you for your contribution.

I left you a review

Please also reset `examples/token-auth-keycloak/data/keycloak/keycloak.trace.db` and `examples/token-auth-keycloak/data/keycloak/keycloak.mv.db`

File: examples/read-only-auth/docker-compose.yml (hunk 1 above)
Comment (GTVolk) [reply examples/read-only-auth/docker-compose.yml:7]: Didn't understand you If you think "/" is mandatory to say docker it is folder mounting, then no, this is folder mounting, docker will create folder if it is not a file present with same name :)

File: examples/token-auth-keycloak/conf/proxy/nginx.conf (hunk 2 above)
Comment (GTVolk) [reply examples/token-auth-keycloak/conf/proxy/nginx.conf:61]: Tab 4 spaces standard for me in all projects) Ok i will use 2 spaces here :)

File: examples/token-auth-keycloak/conf/proxy/nginx.conf (hunk 3 above)
Comment (GTVolk) [reply examples/token-auth-keycloak/conf/proxy/nginx.conf:14]: This is configuration for debugging and see what is happen in logs, i think people want to see this output if they use this example as playground )

File: examples/token-auth-keycloak/conf/proxy/nginx.conf (hunk 4 above)
Comment (GTVolk) [reply examples/token-auth-keycloak/conf/proxy/nginx.conf:50]: Okay, i got it, fix it in next commit 👍

**Comment by GTVolk** [comment]
> Hi, thank you for your contribution.
> 
> I left you a review
> 
> Please also reset `examples/token-auth-keycloak/data/keycloak/keycloak.trace.db` and `examples/token-auth-keycloak/data/keycloak/keycloak.mv.db`

I'll fix your comments, but what you mean about "reset"?) trace.db file is just text file with H2DB exception trace info, this needs to be empty. I saw main DB changet but seems it just activity log written in DB, i don't change that) If you about it and it important i've rollback that in next commit :)

File: examples/read-only-auth/docker-compose.yml (hunk 1 above)
Comment (GTVolk) [reply examples/read-only-auth/docker-compose.yml:7]: Ah ouch sorry, registry.yml didnt dnoticed that

**Comment by GTVolk** [comment]
- Rollback of ```examples/token-auth-keycloak/data/keycloak/keycloak.mv.db```
- Fixed tab 4 spaces to 2 spaces
- Fixed registry.yml

File: examples/issue-75/docker-compose.yml [hunk 5]
```diff
@@ -24,7 +24,7 @@ services:
   s3-server:
     image: minio/minio:RELEASE.2019-04-09T01-22-30Z
     volumes:
-      - ./s3-server-cmd:/bin/s3-server-cmd:ro
+      - ./data/s3-server/s3-server-cmd:/bin/s3-server-cmd:ro
```
Comment (Joxit) [inline examples/issue-75/docker-compose.yml:27]: This may be in `./bin/s3-server/` or `/conf/s3-server/` folder.

**Review by Joxit — CHANGES_REQUESTED** [review]

File: examples/token-auth-keycloak/conf/proxy/nginx.conf (hunk 3 above)
Comment (Joxit) [reply examples/token-auth-keycloak/conf/proxy/nginx.conf]: Some of my users use the examples directly without modifying them. I don't want them to have a development setup in production. You can remove or comment it

File: examples/issue-75/docker-compose.yml (hunk 5 above)
Comment (Joxit) [reply examples/issue-75/docker-compose.yml:27]: You changed the file mode of `examples/issue-75/data/s3-server/s3-server-cmd` from 755 to 644 (removed execution mode for an executable)

**Comment by Joxit** [comment]
Please also rollback `examples/token-auth-keycloak/data/keycloak/keycloak.trace.db`
The keycloak example is pre-configured to work fine, that's why, even if you have changes, you should not commit them.

**Comment by GTVolk** [comment]
> Please also rollback `examples/token-auth-keycloak/data/keycloak/keycloak.trace.db` The keycloak example is pre-configured to work fine, that's why, even if you have changes, you should not commit them.

image
This is content of `examples/token-auth-keycloak/data/keycloak/keycloak.trace.db`. We really need THIS in project ? :)))))

File: examples/issue-75/docker-compose.yml (hunk 5 above)
Comment (GTVolk) [reply examples/issue-75/docker-compose.yml:27]: Windows issue, thanks! i've searching solution for windows Git mess...

**Pull request closed without merging**